library/core/src/main/java/com/google/android/exoplayer2/text/cea/Cea608Decoder.java

```diff
@@ -306,6 +330,38 @@ protected Subtitle createSubtitle() {
     return new CeaSubtitle(cues);
   }
 
+  private boolean isCodeForUnsupportedMode(byte cc1, byte cc2) {
+    // Control codes from 0x01 to 0x0F indicate the beginning of XDS Data
+    if (0x01 <= cc1 && cc1 <= 0x0F) {
+      return true;
+    }
+
+    // 2 commands switch to TEXT mode.
+    if (((cc1 & 0xF7) == 0x14) // first byte must be 0x14 or 0x1C based on channel
+        && (cc2 == CTRL_TEXT_RESTART || cc2 == CTRL_RESUME_TEXT_DISPLAY)) {
+      return true;
+    }
+
+    return false;
+  }
+
+  private static boolean isControlCodeSwitchingToCaptionMode(byte cc1, byte cc2) {
+    if ((cc1 & 0xF7) != 0x14) { // Matching commands must have the CC1 value: 0|0|0|1|CH|1|0|0 where CH is the channel bit
+      return false;
+    }
+
+    switch (cc2) {
+      case CTRL_END_OF_CAPTION:
```

Comment:
See C.16 of the CEA608 standard.
>  EDM or ENM command does not instruct the decoder to treat subsequent data as caption data
...
There are six caption-specific commands which do terminate a Text Mode data stream: EOC, RCL, RDC, RU2, RU3, and RU4.